src/routers/StackRouter.js

```diff
@@ -92,11 +92,16 @@ export default (routeConfigs, stackConfig = {}) => {
       ...(action.params || {}),
       ...(initialRouteParams || {}),
     };
+    const { initialRouteKey } = stackConfig;
     route = {
       ...route,
       ...(params ? { params } : {}),
       routeName: initialRouteName,
-      key: action.key || generateKey(),
+      key:
+        action.key ||
+        (initialRouteKey && initialRouteKey === 'initialRouteName'
```

Comment:
we shouldn't special case this, if you want it to be the same as the initial route name then just pass in a key that is the same as the initial route name